x-pack/plugins/stack_connectors/public/connector_types/inference/additional_options_fields.tsx

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useCallback } from 'react';
+import { css } from '@emotion/react';
+
+import {
+  EuiFormRow,
+  EuiSpacer,
+  EuiTitle,
+  EuiAccordion,
+  EuiFieldText,
+  useEuiTheme,
+  EuiTextColor,
+  EuiButtonGroup,
+  EuiPanel,
+  EuiHorizontalRule,
+  EuiButtonEmpty,
+  EuiCopy,
+  EuiButton,
+  useEuiFontSize,
+} from '@elastic/eui';
+import {
+  getFieldValidityAndErrorMessage,
+  UseField,
+  useFormContext,
+} from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { fieldValidators } from '@kbn/es-ui-shared-plugin/static/forms/helpers';
+import { ConnectorConfigurationFormItems } from '../lib/dynamic_config/connector_configuration_form_items';
+import * as i18n from './translations';
+import { DEFAULT_TASK_TYPE } from './constants';
+import { ConfigEntryView } from '../lib/dynamic_config/types';
+import { Config } from './types';
+import { TaskTypeOption } from './helpers';
+
+// Custom trigger button CSS
+const buttonCss = css`
+  &:hover {
+    text-decoration: none;
+  }
+`;
+
+interface AdditionalOptionsConnectorFieldsProps {
+  config: Config;
+  readOnly: boolean;
+  isEdit: boolean;
+  optionalProviderFormFields: ConfigEntryView[];
+  onSetProviderConfigEntry: (key: string, value: unknown) => Promise<void>;
+  onTaskTypeOptionsSelect: (taskType: string, provider?: string) => Promise<void>;
+  selectedTaskType?: string;
+  taskTypeFormFields: ConfigEntryView[];
+  taskTypeSchema: ConfigEntryView[];
+  taskTypeOptions: TaskTypeOption[];
+}
+
+// const InferenceAPIConnectorFields
+
+export const AdditionalOptionsConnectorFields: React.FC<AdditionalOptionsConnectorFieldsProps> = ({
+  config,
+  readOnly,
+  isEdit,
+  taskTypeOptions,
+  optionalProviderFormFields,
+  taskTypeFormFields,
+  taskTypeSchema,
+  selectedTaskType,
+  onSetProviderConfigEntry,
+  onTaskTypeOptionsSelect,
+}) => {
+  const xsFontSize = useEuiFontSize('xs').fontSize;
+  const { euiTheme } = useEuiTheme();
+  const { setFieldValue, validateFields } = useFormContext();
+
+  const onSetTaskTypeConfigEntry = useCallback(
+    async (key: string, value: unknown) => {
+      if (taskTypeSchema) {
+        const entry: ConfigEntryView | undefined = taskTypeSchema.find(
+          (p: ConfigEntryView) => p.key === key
+        );
+        if (entry) {
+          if (!config.taskTypeConfig) {
+            config.taskTypeConfig = {};
+          }
```

Comment:
```suggestion
          config.taskTypeConfig ??= {};
```

nit, use [nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment).